### PR TITLE
fix(core): Adjustments for IPv6

### PR DESCRIPTION
**Pull request by the-nic**
Minor adjustment for https://github.com/inverse-inc/sope/pull/61, `NSHost hostWithName` is not working with IPv6 and causes an error on the log to be shown

### CODE REVIEW

**Comment by the-nic** [comment]
Is there interest in this @cgx @extrafu ?

**Comment by the-nic** [comment]
As far as my testing goes, yes. Plus the unit tests I added in this PR, this looks functionally okay. But I'd appreciate review feedback, of course.

What I noticed is that the upgrade scripts will not match an ipv6 config for mysql like `mysql://user:pass@[::1]:3306/sogo/sogo_sessions_folder`

Also, I have tested this on ubuntu, only, though it should work on bsd, too

**Comment by the-nic** [comment]
ping?

File: Tests/Unit/TestNGNetUtlilities.m [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/* TestNGNetUtilities.m - this file is part of SOGo
+ *
+ * Copyright (C) 2022 Nicolas Höft
```
Comment (cgx) [inline Tests/Unit/TestNGNetUtlilities.m:3]: As you signed the CLA, please replace your name by `Inverse inc.` and remove the `Author` line.

Comment (the-nic) [reply Tests/Unit/TestNGNetUtlilities.m:3]: Yes, with the CLA, I granted inverse an extra license. But that doesnt change the authorship. I can add inverse to the copyright line. But I think keeping my author attribution is just fair

**Pull request merged**